File: hummingdrone.py
def fib(n): 
    if n>=0: 
      if n==0:
        return 0
      elif n==1: 
          return 1
      else: 
          return fib(n-1)+fib(n-2)
    else:
      print("Incorrect number")
      return -1

class UAV:

    # Class attribute
    # name (data type: string)
    # is_landing (data type: boolean, default False)
    # is_takeoff (data type: boolean, default False)
    name = ""
    is_landing = False
    is_takeoff = False 

    # Initializer / Instance attributes
    


    # instance method
    def takeoff(self):
        pass

# Child class (inherits from Dog class)
class Drone(UAV):
    def __init__(self, name):
        self.name = name

    def takeoff(self):
      if(self.is_landing):
        return False
      else:
        is_takeoff = True
        return is_takeoff


if __name__ == '__main__':
  print(fib(6))

File: hummingdrone.cpp
#include <iostream>
using namespace std;



class UAV {
	private:
	String _name;
	bool _is_landing;
	bool _is_takeoff;
	
	public:
	UAV(){
		_is_landing = false;
		_is_takeoff = false;
	}
	bool takeoff(){
		return
	}
	
		
	
}

class Drone : public UAV {
	
	
	public:
	UAV(String name){
		_name = name
	}
	
	bool takeoff(){
		if(_is_landing == true)
			return false;
		else{
			_is_takeoff = true;
			return _is_takeoff;
		}
			
	}
}

int main() 
{
    cout << "Hello, World!";
    return 0;
}